src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -106,7 +122,12 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return _method.ToUpperInvariant().GetHashCode();
+            if (_hashcode == 0)
+            {
+                _hashcode = _method.ToUpperInvariant().GetHashCode();
```

Comment:
Although, this does spark an alternative approach to the implementation... instead of the private constructor with the awkward `initializeHashCode` parameter, the optimization could be implemented like this:

```c#
// no change to the existing constructor; no additional private constructor

...

public override int GetHashCode()
{
    if (_hashcode == 0)
    {
        switch (_method)
        {
            case "GET":
            case "PUT":
            case "POST":
            case "DELETE":
            case "HEAD":
            case "OPTIONS":
            case "TRACE":
                // _method already uppercase, so no need to use ToUpperInvariant()
                _hashcode = _method.GetHashCode();
                break;

            default:
                _hashcode = _method.ToUpperInvariant().GetHashCode();
                break;
        }
    }

    return _hashcode;
}
```